src/edu/csus/ecs/pc2/ui/MultipleDataSetPane.java

```diff
@@ -572,12 +584,33 @@ public ProblemDataFiles loadDataFiles(Problem aProblem, ProblemDataFiles files,
             throw new RuntimeException("Mismatch: expecting the same number of  '" + dataExtension + "'  and  '" + answerExtension + "'  files in " + dataFileBaseDirectory + "\n (found "
                     + inputFileNames.length + "  '" + dataExtension + "'  files vs. " + answerFileNames.length + "  '" + answerExtension + "'  files)");
         }
-
+        
         SerializedFile[] inputFiles = Utilities.createSerializedFiles(dataFileBaseDirectory, inputFileNames, externalDataFiles);
         SerializedFile[] answertFiles = Utilities.createSerializedFiles(dataFileBaseDirectory, answerFileNames, externalDataFiles);
+
+        SerializedFile [] existingFiles = files.getJudgesDataFiles();
+        
+        if (existingFiles != null && existingFiles.length > 0){
+            /**
+             * Existing files present, concatenate files
+             */
+            
+            Object[] newFiles = Utilities.concatArrays(existingFiles, inputFiles);
+            inputFiles = (SerializedFile[]) newFiles;
+        }
+        
+        existingFiles = files.getJudgesAnswerFiles();
+        
+        if (existingFiles != null && existingFiles.length > 0){
+            /**
+             * Existing files present, concatenate files
+             */
+            Object[] newFiles = Utilities.concatArrays(existingFiles, answertFiles);
+            answertFiles = (SerializedFile[]) newFiles;
```

Comment:
Is there some reason for the unusual variable name `answertFiles`?   I suggest refactoring it to be `answerFiles`.